pkg/cloudprovider/providers/gce/gce_annotations.go

```diff
@@ -58,8 +62,12 @@ func GetLoadBalancerAnnotationType(service *v1.Service) (LoadBalancerType, bool)
 // GetLoadBalancerAnnotationBackendShare returns whether this service's backend service should be
 // shared with other load balancers. Health checks and the healthcheck firewall will be shared regardless.
 func GetLoadBalancerAnnotationBackendShare(service *v1.Service) bool {
-	l, exists := service.Annotations[ServiceAnnotationILBBackendShare]
-	if exists && l == "true" {
+	if l, exists := service.Annotations[ServiceAnnotationILBBackendShare]; exists && l == "true" {
+		return true
+	}
+
+	// Check for deprecated annotation key
+	if l, exists := service.Annotations[deprecatedServiceAnnotationILBBackendShare]; exists && l == "true" {
```

Comment:
This is just a helper function, so the best we could do is glog a warning. However, I haven't seen that for other deprecated annotations: https://github.com/kubernetes/kubernetes/blob/e7b1814863c9b5bb3369deedfe5dccaf7837e27d/pkg/api/service/util.go#L73-L95